raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
(Suddenly realizing it would be better if my first pull request to an OT repo were not trying to punch holes in these abstractions.)  I would ask @bensigelman to fill in some motivation here, since I'm not sure I'll get the language right.